docs/site/Parsing-requests.md

```diff
@@ -0,0 +1,202 @@
+---
+lang: en
+title: 'Parsing requests'
+keywords: LoopBack 4.0, LoopBack 4
+tags:
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Parsing-requests.html
+summary:
+---
+
+## Parsing Requests
+
+This is an action in the default HTTP sequence, it parses arguments from an
+incoming request and uses them as inputs to invoke the corresponding controller
+method.
+
+This action contains 3 steps:
+
+- Parses arguments from request query, body, path and header according to the
+  operation's OpenAPI specification.
+- Coerces parameters from string to its corresponding JavaScript run-time type.
+- Performs validation on the parameters and body data.
+
+### Parsing Raw Data
+
+The code below defines a typical endpoint by decorating a controller method with
+rest decorators.
+
+```ts
+class TodoController {
+  constructor(@repository(TodoRepository) protected todoRepo: TodoRepository) {}
+
+  @put('/todos/{id}')
+  async replaceTodo(
+    @param.path.number('id') id: number,
+    @requestBody() todo: Todo,
+  ): Promise<boolean> {
+    return await this.todoRepo.replaceById(id, todo);
+  }
+}
+```
+
+An operation specification will be generated in-memory to describe it, and raw
+data are parsed from request according to the specification. In the example
```

Comment:
another nitpick: `and raw data are parsed from request` -> `and raw data is parsed from the request`